src/content/en/fundamentals/progressive-web-apps/prpl.md

```diff
@@ -0,0 +1,227 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+description: 
+
+{# wf_updated_on: 2016-09-28 #}
+{# wf_published_on: 2016-09-28 #}
+
+# The PRPL Pattern {: .page-title }
+
+{% include "web/_shared/contributors/addyosmani.html" %}
+
+Dogfood: PRPL is a pattern we believe makes sense for applications that can
+be granularly served. It is currently considered experimental, however we plan on publishing 
+more data on where it works well in the future. 
+
+## Introduction
+
+The web was not initially designed to build modern, highly-interactive applications.
+
+Over the years, we've used JavaScript to try filling this gap. We've invented clever models for structuring
+our apps, built libraries and frameworks to implement them, and layered lots of other useful
+features on top. We've also constructed elaborate tool chains to build and serve and bundle and split
+our JavaScript code. These innovations have filled the gaps in the web platform, but they've also
+added cost in the form of complexity, developer lock-in and JavaScript overhead.
+
+This overhead was a fine tradeoff for the desktop web, where big beefy machines with
+fast CPUs are connected via wire or WiFi to the internet. But **the advent and reach
+of the _mobile_ web fundamentally changes the equation**. Flaky, slow data connections are
+the norm. Underpowered CPUs and tight on-device memory are the norm. And as the next
+billion users come online, many of them are coming online with the most challenging
+combinations of data and device.
+
+Fortunately, the web platform is evolving to meet these challenges:
+
+* Thanks to
+[Service Worker](/web/fundamentals/primers/service-worker/), developers
+have a way to reliably cache these granular components, to ensure their web applications can
+still function and perform reliably even in flaky network conditions.
+
+* Thanks to
+[HTTP/2](https://http2.github.io/), developers have a way to efficiently deliver granular
+dependencies over the network without incurring the high overhead of multiple round-trips.
+
+* Thanks to
+[Web Components](http://webcomponents.org/), developers have a way to extend HTML itself:
+to unlock the powerful component model that's baked directly into the browser and build an
+application out of granular, low-overhead components.
+
+## Towards granular serving
```

Comment:
Imo, the context is useful. With the current PRPL doc on pp.org, anytime I reference it I need to first remind folks why this pattern is interesting, real-world device testing, 3G, interactivity etc before we get to the meat of it. It could probably be trimmed down some.